ospfd/ospf_zebra.c

```diff
@@ -66,23 +66,44 @@ static int
 ospf_router_id_update_zebra (int command, struct zclient *zclient,
 			     zebra_size_t length, vrf_id_t vrf_id)
 {
-  struct ospf *ospf;
+  struct ospf *ospf = NULL;
   struct prefix router_id;
   zebra_router_id_update_read(zclient->ibuf,&router_id);
 
   if (IS_DEBUG_OSPF (zebra, ZEBRA_INTERFACE))
     {
       char buf[PREFIX2STR_BUFFER];
       prefix2str(&router_id, buf, sizeof(buf));
-      zlog_debug("Zebra rcvd: router id update %s", buf);
+      zlog_debug("Zebra rcvd: router id update %s vrf %s id %u",
+        buf, ospf_vrf_id_to_name (vrf_id), vrf_id);
     }
 
   router_id_zebra = router_id.u.prefix4;
 
-  ospf = ospf_lookup ();
+  ospf = ospf_lookup_by_vrf_id (vrf_id);
+  if (!ospf)
+    {
+      if (IS_DEBUG_OSPF_TRACE)
```

Comment:
The OSPF_TRACE is for internal debug activity wherever appropriate also used ZEBRA_INTERFACE